src/CMakeLists.txt

```diff
@@ -0,0 +1,205 @@
+# set up build script
```

Comment:
Don't like long files with multiple responsibilities, they are harder to understand. I rather prefer to combine small and simple ones. Also this makes the code more self documented.

Reply:
Just a general remark here (not implying anything about this particular file): The argument for long, linear "script" type files is that as a newcomer you can read them from top to bottom and have a pretty good understanding of what the code is doing. Contrast this with code scattered around in many files - it can become hard to see the "big picture" flow of execution quickly if not managed properly.

Reply:
This script is quite a bit too long for my taste, too, I agree on that with @azubieta. We should be able to make that script a bit snappier. However, the point of this PR is mainly to restructure the repository layout.